### ISSUE TITLE
Manage errors in the UI

### ISSUE THREAD

**Issue by Dexterp37** [mozilla-ion/ion-core-addon#37]
As highlighted in [this comment](https://github.com/mozilla-ion/ion-core-addon/pull/30#discussion_r502587660) errors that bubble up from the core addon require handling: do we suppress them in the background script or do we want to let user act on them, if they are actionable?

**Comment by Dexterp37** [comment]
@hamilton what's your take on this? For example, what happens if we cannot fetch the list of available studies? Or if user tries to enroll in a study with a "fake id"?

**Comment by hamilton** [comment]
Yep we need to catch these and provide some kind of error message in the UI for these error cases. We will probably handle these error cases using a [notification bar](https://protocol.mozilla.org/patterns/molecules/notification-bar.html) once the mocks begin. In the meantime we can put up something very inexpensive.

**Comment by hamilton** [comment]
We chatted about how to proceed here. The goal would be to have in the persistent store some way to have `userNotifications`, which would be an array of objects of some kind that have notification levels (info, warning, error maybe), some sort of notification code or text, and a timestamp of when the notification entered the store. The UI will then respond accordingly.

**Comment by marniepw** [comment]
When you join, tell the user. When/if the join/fail happens, tell the user.
